src/main/java/org/apache/mesos/config/RepairConfiguration.java

```diff
@@ -0,0 +1,93 @@
+package org.apache.mesos.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Represents the repair configuration in JSON.
+ */
+public class RepairConfiguration {
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+        return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        RepairConfiguration that = (RepairConfiguration) o;
+
+        if (gracePeriodMins != that.gracePeriodMins) {
+            return false;
+        }
+        if (repairDelaySecs != that.repairDelaySecs) {
+            return false;
+        }
+        return enableReplacement == that.enableReplacement;
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = gracePeriodMins;
+        result = 31 * result + repairDelaySecs;
+        result = 31 * result + (enableReplacement ? 1 : 0);
+        return result;
+    }
+
+    @JsonProperty("repairInPlaceGracePeriodMins")
+    private int gracePeriodMins;
+    @JsonProperty("minDelayBetweenRepairsSecs")
+    private int repairDelaySecs;
+    @JsonProperty("enableReplacement")
+    private boolean enableReplacement;
+
+    public  RepairConfiguration() {}
+
+    @JsonCreator
+    public RepairConfiguration(
+            @JsonProperty("repairInPlaceGracePeriodMins") int gracePeriodMins,
+            @JsonProperty("minDelayBetweenRepairsSecs") int repairDelaySecs,
+            @JsonProperty("enableReplacement") boolean enableReplacement) {
+        this.gracePeriodMins = gracePeriodMins;
+        this.repairDelaySecs = repairDelaySecs;
+        this.enableReplacement = enableReplacement;
+    }
+
+    public boolean isEnableReplacement() {
+        return enableReplacement;
+    }
+
+    @JsonProperty("enableReplacement")
+    public void setEnableReplacement(boolean enableReplacement) {
+        this.enableReplacement = enableReplacement;
+    }
+
+    public int getRepairDelaySecs() {
```

Comment:
Yet another reason I don't like YAML for config.  Oh well.  If it's too hard to have both config in YAML and have classes that reasonable signatures for internal consumption, we can stay with the current approach.  To be clear, my proposal was to essentially leave what you have here alone, and add something which provides a better interface for internal consumption, and translate between the two as necessary.